ESP8266/ESP8266.cpp

```diff
@@ -617,6 +623,11 @@ int32_t ESP8266::_recv_tcp_passive(int id, void *data, uint32_t amount, uint32_t
 {
     int32_t ret;
 
+    // return immediately if no data is available
+    if (_sock_i[id].tcp_data_avbl == 0) {
+        return 0;
```

Comment:
Returning zero indicates that the socket is closed - which is not what we want. NSAPI_ERROR_WOULD_BLOCK would be correct.